### ISSUE TITLE
Build Sonic image failed when make configure

### ISSUE THREAD

**Issue by FSSec** [sonic-net/sonic-buildimage#17688]
#### Description

ERROR: failed to solve: process "/bin/sh -c apt-get update && apt-get install -y eatmydata && eatmydata apt-get install -y  ··· "did not complete successfully: exit code: 100
ERROR: failed to solve: sonic-slave-bullseye:7ff9086835e: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
#### Steps to reproduce the issue:
1. git clone https://github.com/Azure/sonic-buildimage.git
2. cd sonic-buildimage/
3. git checkout 202311
4. git pull origin 202311
5. make init
6. make configure PLATFORM=broadcom

#### Describe the results you received:
```
ERROR: failed to solve: process "/bin/sh -c apt-get update && apt-get install -y eatmydata && eatmydata apt-get install -y apt-utils default-jre-headless  openssh-server  curl  wget  unzip  gzip  git  build-essential   libtool         lintian         sudo         dh-make         dh-exec         kmod         libtinyxml2-dev         python-all         python-dev         python-setuptools         python3         python3-pip         libncurses5-dev         texinfo         dh-autoreconf         doxygen         devscripts         git-buildpackage         perl-modules         libclass-accessor-perl         libswitch-perl         libzmq5         libzmq3-dev         uuid-dev         jq         cron         nlohmann-json3-dev         libreadline-dev         texlive-latex-base         texlive-plain-generic         texlive-fonts-recommended         libpam0g-dev         libpam-dev         libcap-dev         imagemagick         ghostscript         groff         libpcre3-dev         gawk         chrpath         libc-ares-dev         libsnmp-dev         libjson-c-dev         libsystemd-dev         python3-ipaddr         libcmocka-dev         python3-all-dev         python3-all-dbg         install-info         logrotate         cdbs         libxml-simple-perl         graphviz         aspell         libgetopt-long-descriptive-perl         libconst-fast-perl         libtemplate-perl         libnamespace-autoclean-perl         libmoose-perl         libmoosex-aliases-perl         bc         fakeroot         build-essential         devscripts         quilt         stgit         sbsigntool         module-assistant         gem2deb         libevent-dev         libglib2.0-dev         python3-all-dev         python3-twisted         phpunit         libbit-vector-perl         openjdk-11-jdk         javahelper         maven-debian-helper         ant         libhttpclient-java         libslf4j-java         libservlet3.1-java         pkg-php-tools         libpcre3         libpcre3-dev         byacc         flex         libglib2.0-dev         bison         expat         libexpat1-dev         dpatch         libdb-dev         libiptc-dev         libxtables-dev \tlibbpf-dev         libtool-bin         libxml2-dev         libusb-1.0-0-dev         libcurl4-openssl-dev         libunwind8-dev         telnet         libc-ares2         libgoogle-perftools4         cpio         squashfs-tools         zip         linux-compiler-gcc-10-x86         linux-kbuild-5.10         libdaemon-dev         libdbus-1-dev         libjansson-dev         libpcap-dev         dnsutils         libusb-dev         libgrpc++1         libabsl-dev         augeas-tools         libyaml-dev         libevent-dev         libjudy-dev         libedit-dev         libnanomsg-dev         python3-stdeb         libjemalloc-dev         liblua5.1-0-dev         lua-bitop-dev          lua-cjson-dev         dkms         cppcheck         clang         pylint         python3-pytest         python3-venv         gcovr         python3-pytest-cov         python3-pytest-cov         python3-parse         default-libmysqlclient-dev         libssl-dev         libperl-dev         libpci-dev         libpci3         libsensors5         libsensors4-dev         libwrap0-dev \tdebhelper         autotools-dev         libbsd-dev         pkg-config         check         docutils-common         libjs-sphinxdoc         libjs-underscore         python3-docutils         python3-jinja2         python3-markupsafe         python3-pygments         python3-roman         python3-sphinx         sphinx-common         python3-sphinx         python3-dev         procmail         libgtest-dev         cmake         libgmock-dev         autoconf-archive         cm-super-minimal         libatm1-dev         libbpf-dev         libelf-dev         libmnl-dev         libselinux1-dev         linuxdoc-tools         lynx         texlive-latex-extra         texlive-latex-recommended         iproute2         texi2html         sharutils         locales         time         man2html-base         libcunit1         libcunit1-dev         shellcheck         bash-completion         dosfstools         qemu-kvm         libvirt-clients         autogen         libopts25-dev         pps-tools         dh-apparmor         librrd8         librrd-dev         rrdtool         liblzo2-dev         libnetfilter-conntrack-dev         libnftnl-dev         xxd         libexplain-dev         libevent-dev         libjsoncpp-dev         swig         device-tree-compiler         autoconf         m4         libxml2-utils         xsltproc         python3-lxml         libexpat1-dev         libcurl4         libcjson-dev         qtbase5-dev                  aspell-en                    libssl-dev                   dbus                         libdbus-1-dev                libdbus-glib-1-2             libdbus-glib-1-dev           libreadline-dev              libncurses5-dev              libpcsclite-dev              docbook-to-man               docbook-utils                libbz2-dev         libboost-dev         libboost-program-options-dev         libboost-system-dev         libboost-serialization1.74-dev         libboost-thread-dev         libboost-atomic-dev         libboost-chrono-dev         libboost-container-dev         libboost-context-dev         libboost-contract-dev         libboost-coroutine-dev         libboost-date-time-dev         libboost-fiber-dev         libboost-filesystem-dev         libboost-graph-parallel-dev         libboost-log-dev         libboost-regex-dev         googletest         libgtest-dev         libgmock-dev         libgcc-10-dev         libcairo2-dev         libdbus-1-dev         libgirepository1.0-dev         libsystemd-dev         pkg-config         libhiredis-dev         libauparse-dev         auditd         dh-elpa         xmlto         rake-compiler         default-jdk         libgoogle-gson-java" did not complete successfully: exit code: 100
Unable to find image 'sonic-slave-bullseye:7ff9086835e' locally
Error response from daemon: pull access denied for sonic-slave-bullseye, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
Error response from daemon: No such container: sonic-slave-bullseye-7ff9086835e
Error response from daemon: No such container: sonic-slave-bullseye-7ff9086835e
Error response from daemon: No such container: sonic-slave-bullseye-7ff9086835e
Error response from daemon: No such image: sonic-slave-bullseye:7ff9086835e
[+] Building 0.5s (3/3) FINISHED                                                                                                                                             docker:default
 => [internal] load build definition from Dockerfile.cleanup                                                                                                                           0.0s
 => => transferring dockerfile: 318B                                                                                                                                                   0.0s
 => [internal] load .dockerignore                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                        0.0s
 => ERROR [internal] load metadata for docker.io/library/sonic-slave-bullseye:7ff9086835e                                                                                              0.5s
------
 > [internal] load metadata for docker.io/library/sonic-slave-bullseye:7ff9086835e:
------
Dockerfile.cleanup:6
--------------------
   4 |     # Copy the cache data to host
   5 |     From scratch as output
   6 | >>> COPY --from=sonic-slave-bullseye:7ff9086835e /cache.tgz cache.tgz
   7 |     
   8 |     # Clean up the cache data
--------------------
ERROR: failed to solve: sonic-slave-bullseye:7ff9086835e: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
[+] Building 0.4s (3/3) FINISHED                                                                                                                                             docker:default
 => [internal] load .dockerignore                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                        0.0s
 => [internal] load build definition from Dockerfile.cleanup                                                                                                                           0.0s
 => => transferring dockerfile: 318B                                                                                                                                                   0.0s
 => ERROR [internal] load metadata for docker.io/library/sonic-slave-bullseye:7ff9086835e                                                                                              0.3s
------
 > [internal] load metadata for docker.io/library/sonic-slave-bullseye:7ff9086835e:
------
Dockerfile.cleanup:9
--------------------
   7 |     
   8 |     # Clean up the cache data
   9 | >>> FROM sonic-slave-bullseye:7ff9086835e as final
  10 |     RUN rm /cache.tgz
  11 |     
--------------------
ERROR: failed to solve: sonic-slave-bullseye:7ff9086835e: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
Error response from daemon: No such image: tmp-sonic-slave-bullseye:7ff9086835e
Error response from daemon: No such container: sonic-slave-bullseye-7ff9086835e
sync: error opening 'sonic-slave-bullseye/vcache/cache.base64': No such file or directory
Error response from daemon: No such container: sonic-slave-bullseye-7ff9086835e
make[1]: *** [Makefile.work:615: configure] Error 1
make[1]: Leaving directory '/data/jovanny/sonic-buildimage'
make: *** [Makefile:94: configure] Error 2
```

#### Describe the results you expected:
Build Pass

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by linqingxuan** [comment]
It seems to be caused by network problems. You can switch to a mirror source closer to you and modify the scripts/build_mirror_config.sh file.

**Comment by volodymyrsamotiy** [comment]
@FSSec to check suggestion added in previous comment.

**Comment by 5eb38fd8cdee** [comment]
I experienced the same issues as @FSSec . 
It is true that the issue I was experiencing was with Debian repositories unable to resolve the hostname:

```
0.451 Err:1 http://debian-archive.trafficmanager.net/debian stretch InRelease
0.451   Temporary failure resolving 'debian-archive.trafficmanager.net'
0.451 Err:2 http://debian-archive.trafficmanager.net/debian stretch-updates InRelease
0.451   Temporary failure resolving 'debian-archive.trafficmanager.net'
0.451 Err:3 http://debian-archive.trafficmanager.net/debian stretch-backports InRelease
0.451   Temporary failure resolving 'debian-archive.trafficmanager.net'
0.451 Err:4 http://packages.trafficmanager.net/debian/debian stretch InRelease
0.451   Temporary failure resolving 'packages.trafficmanager.net'
0.451 Err:5 http://packages.trafficmanager.net/debian/debian stretch-updates InRelease
0.451   Temporary failure resolving 'packages.trafficmanager.net'
0.451 Err:6 http://packages.trafficmanager.net/debian/debian stretch-backports InRelease
0.451   Temporary failure resolving 'packages.trafficmanager.net'
0.451 Err:7 http://packages.trafficmanager.net/debian/debian-security stretch_updates InRelease
0.451   Temporary failure resolving 'packages.trafficmanager.net'
0.451 Err:8 http://debian-archive.trafficmanager.net/debian-security stretch/updates InRelease
0.451   Temporary failure resolving 'debian-archive.trafficmanager.net'
0.456 Reading package lists...
0.472 W: Failed to fetch http://debian-archive.trafficmanager.net/debian/dists/stretch/InRelease  Temporary failure resolving 'debian-archive.trafficmanager.net'
0.472 W: Failed to fetch http://debian-archive.trafficmanager.net/debian/dists/stretch-updates/InRelease  Temporary failure resolving 'debian-archive.trafficmanager.net'
0.472 W: Failed to fetch http://debian-archive.trafficmanager.net/debian/dists/stretch-backports/InRelease  Temporary failure resolving 'debian-archive.trafficmanager.net'
0.472 W: Failed to fetch http://packages.trafficmanager.net/debian/debian/dists/stretch/InRelease  Temporary failure resolving 'packages.trafficmanager.net'
0.472 W: Failed to fetch http://packages.trafficmanager.net/debian/debian/dists/stretch-updates/InRelease  Temporary failure resolving 'packages.trafficmanager.net'
0.472 W: Failed to fetch http://packages.trafficmanager.net/debian/debian/dists/stretch-backports/InRelease  Temporary failure resolving 'packages.trafficmanager.net'
0.472 W: Failed to fetch http://debian-archive.trafficmanager.net/debian-security/dists/stretch/updates/InRelease  Temporary failure resolving 'debian-archive.trafficmanager.net'
0.472 W: Failed to fetch http://packages.trafficmanager.net/debian/debian-security/dists/stretch_updates/InRelease  Temporary failure resolving 'packages.trafficmanager.net'
0.472 W: Some index files failed to download. They have been ignored, or old ones used instead.
```

In parallel, the secondary/fallback option failed as well where if the build failed, download the sonic-slave-* image from docker.io ... which would result in having the image non-existent:

```
docker search sonic-slave-buster
NAME                                   DESCRIPTION   STARS     OFFICIAL
vcmy/sonic-slave-buster                              0         
kerongbaby/sonic-slave-buster                        0         
vcmy/sonic-slave-buster-vcmy                         0         
mtanlee/sonic-slave-buster                           0         
redrattus/sonic-slave-buster                         0         
kerongbaby/sonic-slave-buster-wangyc                 0         
kerongbaby/sonic-slave-buster-202205                 0         

docker search sonic-slave-bullseye
NAME                                     DESCRIPTION   STARS     OFFICIAL
mtanlee/sonic-slave-bullseye                           0         
kerongbaby/sonic-slave-bullseye-202205                 0         
kerongbaby/sonic-slave-bullseye                        0         

docker search sonic-slave-jessie
NAME                                 DESCRIPTION   STARS     OFFICIAL
vcmy/sonic-slave-jessie-vcmy                       0         
vcmy/sonic-slave-jessie                            0         
mtanlee/sonic-slave-jessie                         0         
a963383404/sonic-slave-jessie-vcmy                 0         
a963383404/sonic-slave-jessie                      0         
x4fx4e/sonic-slave-jessie-minty                    0         
```

The issue is with Docker not being able to resolve DNS entries (i.e.: during apt-get update) was resolved thanks to this post:

https://stackoverflow.com/a/68199803

Basically create a " /etc/docker/daemon.json " file and add DNS entires within that JSON file. Strange issue, not sure why this workaround had to be applied but it allowed me to move beyond this error (and onto a new error :) )
